### PR TITLE
feat(npm-check): integrate npm check to track outdated deps

### PR DESCRIPTION
**Pull request by sk91**
#267 deps checks

- [x] run npm check and do minor upgrades 
- [x] add yarn command to analyze deps `yarn check-deps`
- [x] add outdated packages check on precommit
- [x] clean up packages
example precommit message:
```
[18:48:41] gulp-npm-check
 Out of date packages:
        @ledgerhq/hw-transport-node-hid (installed: 4.16.0, latest: 4.19.1)
        @ledgerhq/web3-subprovider (installed: 4.15.0, latest: 4.19.0)
        redux (installed: 3.7.2, latest: 4.0.0)
```

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const tmplSrc = path.resolve(__dirname, './src/renderer/templates/**/*.pug');
 const tmplDest = path.resolve(__dirname, './src/renderer/angular/');
 const scssSrc = path.resolve(__dirname, './static/stylesheets/scss/main.scss');
 const scssDest = path.resolve(__dirname, './static/stylesheets/css');
+const npmcheck = require('gulp-npm-check');
```
Comment (rodrigopavezi) [inline gulpfile.js]: @sk91 can't we just use the https://www.npmjs.com/package/npm-check in a script on the package.json? We are trying to remove the dependency on gulp from the wallet and leave only webpack.  cheers

Comment (sk91) [reply gulpfile.js]: @rodrigopavezi  you are correct, fixing

**Comment by sk91** [comment]
@rodrigopavezi  fixed

**Comment by sk91** [comment]
@rodrigopavezi @dfcowell ready for merge

**Pull request merged**